assessment.rb

```diff
@@ -0,0 +1,69 @@
+module CompanyConcerns
+  # All methods related to Payment for a company
+  module Payment
+    extend ActiveSupport::Concern
+
+    class_methods do
+      def generate_todays_invoice!
+        # generating today's invoice means a lot of stuff
+        # example: if today is the company's payment day, we need to pay all the amount
+        # if today is a normal day, we need to see if they have new seats
+
+        handle_save do
+          active_employees = employees.where(active: true).count
+          current_billing_date = self.current_billing_date
+          next_billing_date = self.next_billing_date
+
+          if DateTime.now.day == billing_cycle
+            # means it's the new billing cycle
+            payment_amount = payment_for_employees(0, active_employees)
+
+            if payment_amount.positive?
+              if in_trial? && trial_ends_at.to_date > DateTime.now.to_date
+                # we do nothing
+                # in this case, the user is in trial, and the trial did not end yet
+                # however, if he's in trial and the trial ends today, we need to bill him
+              else
+                company_invoices.create!(
+                  date_from: DateTime.now,
+                  date_to: next_billing_date + 1.month,
+                  seats: active_employees,
+                  expected_price: payment_amount,
+                  total_price: payment_amount,
+                  currency: currency
+                )
+              end
```

Comment:
If `in_trial?` is just a boolean, I believe `trial_ends_at` would be sufficient to determine whether or not the company is in trial period. Also I would assume that `in_trial?` checks whether the trial has ended or not.

One way to simplify this would be,
```suggestion
              if payment_ && trial_ends_at.to_date > DateTime.now.to_date
                # we do nothing
                # in this case, the user is in trial, and the trial did not end yet
                # however, if he's in trial and the trial ends today, we need to bill him
              else
                company_invoices.create!(
                  date_from: DateTime.now,
                  date_to: next_billing_date + 1.month,
                  seats: active_employees,
                  expected_price: payment_amount,
                  total_price: payment_amount,
                  currency: currency
                )
              end
```